javascript/organizing_your_javascript_code/es6_modules.md

```diff
@@ -1,183 +1,242 @@
+</div>
 
-Here is an example with named exports, which you will most often use when you have multiple values to export in a module:
+#### Default exports
+
+In contrast to named exports, a file can only default export a single thing. Something exported this way does not have a name attached to it, so when you import it somewhere, you can decide what name to give it.
+
+To export something from a file as a default export, we can also do it inline by prepending `export default` to the appropriate declaration, or we can export it at the end of the file, this time *without* any curly braces. Again, either way is perfectly fine. Note that if you do an inline default export, if you would normally declare with `let` or `const`, the `default` keyword *replaces* `let`/`const`.
 
 ```javascript
-// a file called myModule.js
-const functionOne = () => 'ONE';
-const functionTwo = () => 'TWO';
+// one.js
+export default greeting = "Hello, Odinite!";
 
-export {
-  functionOne,
-  functionTwo
-};
+// Or on a separate line
+const greeting = "Hello, Odinite!";
+export default greeting;
 ```
 
-And to import them:
+Now in our `two.js`, we can default import that string. Remember, since we're importing something that was default exported, we can name it whatever we want. Even though the variable was called `greeting` in `one.js`, we don't have to call it that in `two.js` if we don't want to. When default importing, we don't use curly braces, which are for named importing.
 
 ```javascript
-// index.js in /src folder
-import {functionOne, functionTwo} from './myModule';
+// two.js
+import helloOdinite from "./one.js";
+
+console.log(helloOdinite); // "Hello, Odinite!"
 ```
 
-Using this pattern gives you the freedom to only import the functions you need in the various files of your program. So it's perfectly fine to only import `functionOne` if that's the only one you need.
+You can use both default and named exports in the same file. Confusingly enough, there isn't really a universally agreed-upon rule for when to use either, outside of the fact that a file can have multiple named exports but only one default export. When it comes to only needing to export a single thing from a module, some people prefer using a default export whereas some prefer using a single named export. Both work so use whatever you prefer, or in the future, when working in a team, whatever your team prefers if they prefer a certain system.
+
+```javascript
+// one.js
+export default greeting = "Hello, Odinite!";
+export const farewell = "Bye bye, Odinite!";
+```
+
+```javascript
+// two.js
+import greeting, { farewell } from "./one.js";
+```
+
+### CommonJS
```

Comment:
Kind of a general note, I feel like the content of this lesson shifts pretty hard when we start covering npm.

Would it maybe make more sense to have one lesson that covers the JS module types, and another that just covers npm and packaging?

Reply:
Just to clarify, are you suggesting another lesson split to accommodate this?
i.e.
```txt
ES6 Modules -> npm -> Webpack -> Project: Restaurant Page -> Revisiting Webpack
```
I'd be completely fine with this and agree it'd help keep the lessons more focused. I just kept the npm stuff at the end of ESM since it was maintaining the current structure, with only the new 2nd webpack lesson to be added.

Reply:
Yep!

I think whether we cover both ESM and CJS in one lesson or put CJS in with npm could be a judgment call.

I don't think either would really be _wrong_, so I would probably lean toward just putting CJS with whichever lesson is shorter after the split.

Reply:
Awesome - I'll look into splitting out a new npm lesson between ESM and Webpack, which should make the flow a little easier to deal with. I'll look into the other suggestions in the meantime as well.